temporal-server/rockcraft.yaml

```diff
@@ -0,0 +1,25 @@
+name: temporal-server
+# see https://documentation.ubuntu.com/rockcraft/en/1.11.0/explanation/bases/
+# for more information about bases and using 'bare' bases for chiselled rocks
+base: ubuntu@24.04 # the base environment for this rock
+version: '1.23' # just for humans. Semantic versioning is recommended
+summary: Rock for Temporal Server # 79 char long summary
+description: |
+    Temporal is a durable execution platform that enables developers to build scalable applications without sacrificing productivity or reliability. The Temporal server executes units of application logic called Workflows in a resilient manner that automatically handles intermittent failures, and retries failed operations.
+
+platforms: # the platforms this rock should be built on and run on
+    amd64:
+
+parts:
+    temporal:
+        plugin: go
+        source: https://github.com/temporalio/temporal
+        source-subdir: cmd/server 
+        source-type: git
+        source-tag: release/v1.23.x
```

Comment:
Should we use the tag instead of the branch? There is no `release/v1.23.x` tag, but there is a [1.23.1](https://github.com/temporalio/temporal/tree/v1.23.1) tag.